diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_cik.c b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_cik.c
index 05f3ac2eaef9..23b360274df1 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_cik.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_cik.c
@@ -99,6 +99,7 @@ static void init_mqd(struct mqd_manager *mm, void **mqd,

 	m->header = 0xC0310800;
 	m->compute_pipelinestat_enable = 1;
+	m->compute_perfcount_enable = 1;
 	m->compute_static_thread_mgmt_se0 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se1 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se2 = 0xFFFFFFFF;
diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v10.c b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v10.c
index 1695dd78ede8..1541c056fd12 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v10.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v10.c
@@ -99,6 +99,7 @@ static void init_mqd(struct mqd_manager *mm, void **mqd,

 	m->header = 0xC0310800;
 	m->compute_pipelinestat_enable = 1;
+	m->compute_perfcount_enable = 1;
 	m->compute_static_thread_mgmt_se0 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se1 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se2 = 0xFFFFFFFF;
diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v11.c b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v11.c
index 3c0ae28c5923..4d2fe0faa0e6 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v11.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v11.c
@@ -141,6 +141,7 @@ static void init_mqd(struct mqd_manager *mm, void **mqd,

 	m->header = 0xC0310800;
 	m->compute_pipelinestat_enable = 1;
+	m->compute_perfcount_enable = 1;

 	m->compute_static_thread_mgmt_se0 = wa_mask;
 	m->compute_static_thread_mgmt_se1 = wa_mask;
diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v12.c b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v12.c
index 565858b9044d..0c038690ccc6 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v12.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v12.c
@@ -109,6 +109,7 @@ static void init_mqd(struct mqd_manager *mm, void **mqd,

 	m->header = 0xC0310800;
 	m->compute_pipelinestat_enable = 1;
+	m->compute_perfcount_enable = 1;
 	m->compute_static_thread_mgmt_se0 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se1 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se2 = 0xFFFFFFFF;
diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c
index f2dee320fada..f24357a20f97 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c
@@ -171,6 +171,7 @@ static void init_mqd(struct mqd_manager *mm, void **mqd,

 	m->header = 0xC0310800;
 	m->compute_pipelinestat_enable = 1;
+	m->compute_perfcount_enable = 1;
 	m->compute_static_thread_mgmt_se0 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se1 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se2 = 0xFFFFFFFF;
diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_vi.c b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_vi.c
index c1fafc502515..e7b00734ad58 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_vi.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_vi.c
@@ -102,6 +102,7 @@ static void init_mqd(struct mqd_manager *mm, void **mqd,

 	m->header = 0xC0310800;
 	m->compute_pipelinestat_enable = 1;
+	m->compute_perfcount_enable = 1;
 	m->compute_static_thread_mgmt_se0 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se1 = 0xFFFFFFFF;
 	m->compute_static_thread_mgmt_se2 = 0xFFFFFFFF;
